### ISSUE TITLE
Mongo Support

### ISSUE THREAD

**Issue by timwford** [awtkns/fastapi-crudrouter#32]
Probably use ODMantic: https://github.com/art049/odmantic

How would you test the implementation? mock a mongo db?

**Comment by awtkns** [comment]
I agree.  We should support mongo support mongo.  I think it would be fairly easy with odmantic! Alternatively just the motor and pydantic could work.

**Comment by awtkns** [comment]
>How would you test the implementation? mock a mongo db?

Testing mongo could be done with docker potentially

**Comment by n0nvme** [comment]
+1 for motor and pydantic instead of odmantic

**Comment by tclasen** [comment]
Hello! I'm resurrecting this thread to recommend https://roman-right.github.io/beanie/ as the ODM for this ticket.

**Comment by tclasen** [comment]
> Hello! I'm resurrecting this thread to recommend https://roman-right.github.io/beanie/ as the ODM for this ticket.

Also, there is now support for mocking out motor during unit tests that don't really need a full running mongodb: https://github.com/michaelkryukov/mongomock_motor

**Comment by michele-tasca** [comment]
Hello everyone! Any news on this ?

Is this (using crudrouter with mongodb) currently possible in any way (even hacky), or is it just not supported?

**Comment by tclasen** [comment]
I was able to get it working with Beanie in a fork, I just didn't finish out the typing to get it to pass mypy. It is very possible.

**Comment by michele-tasca** [comment]
I could definitely use that fork!😁

If typing requires some work, I'd be happy to work on it if you want, so we can open a pull request.

Idk if this repo is still mantained, but the docs say the list of supported DBs is supposed to grow so..

**Comment by dclimber** [comment]
@tclasen where can one find the fork? Could you please send the link?

**Comment by dclimber** [comment]
It is indeed possible to make it work with Beanie — working on it.

**Comment by tclasen** [comment]
Hey, @dclimber , sorry I missed your original tag. I'd happily assist you with building this out. Maybe even @killswitch-GUI would want to take a look at this as well.

**Comment by dclimber** [comment]
@tclasen cool! I've made it work — the `BeanieCRUDRouter`, based it on `OrmarCRUDRouter`, as they have `pydatic`'s `BaseModel` in common. Manual tests during 'experiments' succeeded. 
Now I need to write the actual tests, and I guess it's ready :)

**Comment by micoloth** [comment]
Hi guys! Are there news on this?

@dclimber  A very rough draft would still be very useful for me at this moment! 

I can contribute and finish working on it if needed

**Comment by dclimber** [comment]
@micoloth I have it working with ODMantic, but need to adjust tests, so that they pass.
I guess I can publish it into a branch

**Comment by micoloth** [comment]
Thank you for your work!

Any of the two branches would be very helpful for me, yes!😌

I can certainly help to work on the tests if that's needed.

Maybe the code can be published as a branch specifying it's WIP work?

**Comment by dclimber** [comment]
@micoloth hey, sorry for a late reply.
Beanie branch: https://github.com/dclimber/fastapi-crudrouter/tree/feature/beanie
ODMantic branch: https://github.com/dclimber/fastapi-crudrouter/tree/feature/odmantic

**Comment by gegnew** [comment]
What's the status here? Any chance these PRs get merged?

**Comment by dclimber** [comment]
@gegnew I needed these features for a project, but it got cancelled, so I no longer work on these MRs. The only thing that is needed is unit tests.
Also there's  an annoying `id` -> `_id` thing to fully solve.

If you're willing to help - feel free to fork and add some tests :)

**Comment by PierroD** [comment]
Hello everyone,

I made a CRUDRouter for MongoDB, with the `id` -> '_id' and vice versa.

Feel free to read the doc and / or try it : 
https://github.com/PierroD/fastapi-crudrouter-mongodb
